src/Render/Program.cs

```diff
@@ -88,11 +88,18 @@ internal static class Program {
 		}
 
 		private static string GetTitle( MarkdownDocument doc ) {
-			var inline = doc.Descendants<HeadingBlock>().Single(
-				h => h.Level == 1
-			).Inline;
-			var titleLiteral = inline.Descendants<LiteralInline>().Single();
-			return titleLiteral.Content.ToString();
+			try {
+				var inline = doc.Descendants<HeadingBlock>().Single(
+					h => h.Level == 1
+				).Inline;
+				var titleLiteral = inline.Descendants<LiteralInline>().Single();
+				return titleLiteral.Content.ToString();
+			}
+			catch ( Exception e) 
+				when (e is InvalidOperationException || e is ArgumentNullException) 
+			{
+				throw new ArgumentException( "Document should have exactly 1 level-1 heading (should be unformatted)", e );

```

Comment:
minor nit: consider replacing the first '1' with the typed out word for readability

```suggestion
				throw new ArgumentException( "Document should have exactly one level-1 heading (should be unformatted)", e );
```